### PR TITLE
chore: update OT to v0.78.0

### PR DESCRIPTION
**Pull request by andrzej-stencel**
This PR contains two commits:
- one that updates OT core to v0.78.2 and OT contrib to v0.78.0
- one that updates the OT update procedure to take into account the difference in OT core and contrib versions.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -95,17 +95,29 @@ check-uniform-dependencies:
...
+		docs/configuration.md \
+		docs/migration.md \
+		pkg/exporter/sumologicexporter/README.md
+	$(SED) -i "s/\(contrib\/\(blob\|tree\)\/v\)$(OT_CONTRIB_VERSION)/\1$(OT_CONTRIB_NEW)/" \
+		README.md \
+		docs/configuration.md \
+		docs/migration.md \
+		pkg/receiver/telegrafreceiver/README.md
+	@find . -type f -name "go.mod" -exec $(SED) -i "s/\(go\.opentelemetry\.io\/collector.*\) v$(OT_CORE_VERSION)$$/\1 v$(OT_CORE_NEW)/" {} \;
+	@find . -type f -name "go.mod" -exec $(SED) -i "s/\(github\.com\/open-telemetry\/opentelemetry-collector-contrib\/.*\) v$(OT_CONTRIB_VERSION)$$/\1 v$(OT_CONTRIB_NEW)/" {} \;
```
Comment (aboguszewski-sumo) [inline Makefile:120]: If you're not going to squash this PR, could we split this change into a separate commit?

Comment (astencel-sumo) [reply Makefile:120]: As I said in the PR description, this PR contains to separate commits that I mean to merge with rebase.

**Review by swiatekm-sumo — APPROVED** [review]
